runtime/interpreter/decode.go

```diff
@@ -171,21 +85,21 @@ func (d *DecoderV5) decodeValue(path []string) (Value, error) {
 		if err != nil {
 			return nil, err
 		}
-		value = BoolValue(v)
+		storable = BoolValue(v)
 
-	case cbor.TextStringType:
-		v, err := d.decoder.DecodeString()
+	case cbor.NilType:
+		err := d.decoder.DecodeNil()
 		if err != nil {
 			return nil, err
 		}
-		value = d.decodeString(v)
+		storable = NilValue{}
 
-	case cbor.NilType:
-		err := d.decoder.DecodeNil()
+	case cbor.TextStringType:
+		v, err := d.decoder.DecodeString()
 		if err != nil {
 			return nil, err
 		}
-		value = NilValue{}
+		storable = stringAtreeValue(v)
```

Comment:
CBOR strings are now decoded to simple atree string values, which are used for composite value field names, and don't have the overhead of full Cadence strings (`*StringValue`) – those are now encoded with a CBOR tag.